content/800-data-platform/050-data-proxy/100-use-data-proxy.mdx

```diff
@@ -119,11 +119,33 @@ You can generate and use Prisma Client for the Data Proxy in the following envir
    <Admonition>
 
    **Note**<br /><br />
-   For security reasons, we strongly recommend that you always use environment variables for your database connection strings and Data Proxy strings. Do not hard-code them into your `schema.prisma` file.<br /><br />
-   For more information, see [Using `.env` files](/guides/development-environment/environment-variables#using-env-files).
+   For security reasons, use environment variables for database connection strings and Data Proxy strings. Do not hard-code them into `schema.prisma`. For more information, see [Using `.env` files](/guides/development-environment/environment-variables#using-env-files).
 
    </Admonition>
 
+1. To introspect your database or perform migrations using the Prisma CLI, a direct database connection is required. If you are using Prisma version 4.1.0 or later, include the `directUrl` property in the [`datasource`](/reference/api-reference/prisma-schema-reference#datasource) block of your Prisma schema.
+
+   ```prisma file=schema.prisma highlight=8;add
+   generator client {
+     provider = "prisma-client-js"
+   }
+   
+   datasource db {
+     provider  = "postgresql"
+     url       = env("DATABASE_URL")
+     directUrl = env("DIRECT_URL")
+   }
+   ```
+
+   Then, add the direct database connection string in the `.env` file.
+
+   ```bash file=.env highlight=2;add
+   DATABASE_URL="prisma://****:****@****"
+   + DIRECT_URL="postgresql://****:****@****"
+   ```
+
+   However, in older versions, you need to overwrite the `DATABASE_URL` environment variable with the direct database connection string when running migrations. Refer to the [Prisma CLI commands with the Data Proxy](https://www.prisma.io/docs/data-platform/data-proxy/prisma-cli-with-data-proxy) documentation for more information.
```

Comment:
This is now content under the headline `Generate Prisma Client for the Data Proxy`. Shouldn't this be under a different top level heading?

Reply:
The content was added to ensure users have a `directUrl` setup when generating the Prisma Client for Data Proxy. As step 1 replaces the database connection, step 2 indicates that migrations won't work without a direct URL and we show how to set that up. I agree it does not go well with the heading, I could move this into a new section below? 
@nikolasburk  @janpio - something like 'Using the direct database connection'

Reply:
Yes, `directUrl` is only for Prisma CLI so maybe that can be used. The current headline uses "Prisma Client", so another one about using Prisma CLI in that situation would make sense.

Reply:
I agree that the headline isn't very accurate any more, nice catch Jan!

Maybe some of these work better now: 

- Configure project to use Data Proxy
- Configure Prisma project to use Data Proxy
- Configure local project to use Data Proxy
- Configure local Prisma project to use Data Proxy

Reply:
Changing the headline to  `Configure Prisma project to use Data Proxy`